### DISCUSSION TITLE
Exclude/Include Sheets based on published and/or approval status

### DISCUSSION THREAD

**Discussion by Chase-Stevison** [ptarmiganlabs/butler-sheet-icons#526] [Ideas]
First of all, thanks for all your work on your suite of Qlik tools! In my use case it would be useful if there were a parameter to include or exclude a sheet from thumbnail generation based on whether the sheet is published, approved, or both. Generally my team is only interested in generating thumbnails for sheets that are both published and approved.

**Comment by mountaindude** [comment]
Good idea - and an obvious one at that 🤦‍♂️.
Should be pretty easy to add, let me see what the Easter bunny brings this year....

**Reply by mountaindude** [reply]
How about a new option `--include-sheet-status` for the `butler-sheet-icons.exe qseow create-sheet-thumbnails` command?

As there are already some options for excluding sheets a decision has to be made if the inclusion options should be considered before the exclude options... I don't think it really matters, or?
Leaning toward including  first, then removing sheets that should be excluded.

**Reply by mountaindude** [reply]
...and same for updating apps in QS Cloud, of course.

**Reply by mountaindude** [reply]
Thinking about it, I think it will be a better idea (easier to both understand and implement) to have two separate options: 

`--include-sheet-status-approved` and `--include-sheet-status-published`

Possible values `true` and `false`, with `true` being default value.

**Reply by mountaindude** [reply]
...or maybe use exclusion here too...

I.e. all sheets are included by default, then any sheets matching the various exclude criteria are excluded. 
May actually be easier and more consistent with current options.

Still some thinking to be done, it seems :)

**Comment by mountaindude** [comment]
Ok, here's the plan:

Add a new option `--exclude-sheet-status <status...>`. 
Possible values are `private`, `published`, `public`. 

Zero or more of the values can be passed to the option.

This will align with the various existing exclude-options, while at the same time allowing handling of a specific type of sheet.

**Reply by Chase-Stevison** [reply]
Sounds great to me! Thanks so much!

**Comment by mountaindude** [comment]
Closing as this feature is now part of BSI.
